pddl/logic/effects.py

```diff
@@ -167,5 +126,5 @@ def __lt__(self, other):
 
 PEffect = Union[Atomic, Not]
 CEffect = Union[Forall, When, OneOf, "PEffect"]
-Effect = Union[AndEffect["CEffect"], CEffect]
-CondEffect = Union[AndEffect["PEffect"], "PEffect"]
+Effect = Union[And, CEffect]
+CondEffect = Union[And, "PEffect"]
```

Comment:
This weaken things a bit, but I couldn't figure out how to optionally get the type in `And`.

Reply:
Yeah...why can't we apply the optional to the `And` here? (Unfortunately, I cannot attempt any trial as I don't have access to my laptop at the moment)

Reply:
It's because the `AndEffect` inherited from `Generic` with a custom effect type: https://github.com/AI-Planning/pddl/blob/main/pddl/logic/effects.py#L28

`And` does not, though, and I couldn't see a clear path to making it work.

Reply:
Going to resolve and merge, but happy to revisit if it becomes necessary. Ultimately what I think this will allow through is the parsing of `(and (p) (or (q) (s)))`, as an effect, which should ideally be an error. That syntax is ok as a precondition (or condition in a conditional effect), but not as an effect.